src/librbd/object_map/InvalidateRequest.h

```diff
@@ -31,9 +31,12 @@ class InvalidateRequest : public AsyncRequest<ImageCtxT> {
 
 protected:
   bool should_complete(int r) override;
-  int filter_return_code(int r) const override{
-    // never propagate an error back to the caller
-    return 0;
+  int filter_return_code(int r) const override {
+    if (r == -ENOENT) {
```

Comment:
I wasn't saying that we shouldn't propagate back the `-ENOENT` error, I was saying we should fix the cause of what was issuing the `-ENOENT` error.